src/Domain/ValueObjects/Wallet.cs

```diff
@@ -0,0 +1,29 @@
+using Domain.Aggregates;
+
+namespace Domain.ValueObjects;
+
+public record Wallet(decimal Balance)
+{
+
+    public decimal Balance { get; private set; } = Balance;
+
+    public bool HasBalance(decimal amount)
+    {
+        if (amount <= 0) throw new ArgumentException(nameof(amount),"amount must be positive and non zero");
```

Comment:
format the file, alt+shift+f so that there are nice and consistent spacing around the whole repo.